ssl/ssl_ciph.c

```diff
@@ -995,14 +995,8 @@ static int ssl_cipher_process_rulestr(const char *rule_str,
             ch = *l;
             buf = l;
             buflen = 0;
-#ifndef CHARSET_EBCDIC
-            while (((ch >= 'A') && (ch <= 'Z')) ||
-                   ((ch >= '0') && (ch <= '9')) ||
-                   ((ch >= 'a') && (ch <= 'z')) ||
-                   (ch == '-') || (ch == '.') || (ch == '='))
-#else
-            while (isalnum(ch) || (ch == '-') || (ch == '.') || (ch == '='))
-#endif
+            while (!(ch & ~127) && (isalnum(ch) || ch == '-' || ch == '.'
+                                    || ch == '='))
```

Comment:
Errr, `!(ch & ~127)` won't fly on EBCDIC, because alnums are actually in upper half. Just leave it as it was... Well, it seems that it would actually be appropriate to cast ch to unsigned char. I mean leave it as it was, but add the cast to isalnum(ch)...